ONSdigital/respondent-home-ui
# Add an additional smoke test for the status of required services

# Motivation and Context

We would like to improve the confidence of our services by adding smoke tests that verify that a service's dependencies are healthy.

NB: The smoke tests are run as part of the concourse pipeline.

# What has changed



Added a check_services method to the app object that uses the health endpoints of the ras/rm services to verify that each are reachable.

Updated the /info endpoint with a `ready` key, which will only be `true` if all the required services are reachable by this service and only if `/info` has the query param `?check` present.

# How to test?



To test locally:

1. Ensure a `.env` file exists in the root directory. ([Example to copy can be found here](https://github.com/ONSdigital/respondent-home-ui/blob/master/tests/test_data/local.env)).
1. `make serve` to start the RH server
1. `RESPONDENT_HOME_URL=http://localhost:9092 pipenv run inv smoke` (with the server running)
1. Try the above command with and without the services running (`make start_services`, `docker kill collection-instrument`, etc.) 

Unit tests:

- `pipenv run inv unittests`
